pytorch_lightning/callbacks/batch_size_finder.py

```diff
@@ -0,0 +1,476 @@
+        elif trainer.state.fn == TrainerFn.VALIDATING:
+            self._dumped_params["num_val_batches"] = trainer.num_val_batches
+            trainer.num_val_batches = [0]
+        elif trainer.state.fn == TrainerFn.TESTING:
+            self._dumped_params["num_test_batches"] = trainer.num_test_batches
+            trainer.num_test_batches = [0]
+        elif trainer.state.fn == TrainerFn.PREDICTING:
+            self._dumped_params["num_predict_batches"] = trainer.num_predict_batches
+            trainer.num_predict_batches = [0]
+
+    def post_early_exit(self, trainer):
+        if trainer.fast_dev_run:
+            return
+
+        # restore the state used to stop the respective loop
+        if trainer.state.fn == TrainerFn.FITTING:
+            trainer.num_training_batches = self._dumped_params["num_training_batches"]
+            loop = trainer.fit_loop
+        if trainer.state.fn == TrainerFn.VALIDATING:
+            trainer.num_val_batches = self._dumped_params["num_val_batches"]
+            loop = trainer.validate_loop
+        if trainer.state.fn == TrainerFn.TESTING:
+            trainer.num_test_batches = self._dumped_params["num_test_batches"]
+            loop = trainer.test_loop
+        if trainer.state.fn == TrainerFn.PREDICTING:
+            trainer.num_predict_batches = self._dumped_params["num_predict_batches"]
+            loop = trainer.predict_loop
+
+        loop.load_state_dict(self._dumped_params["loop_state_dict"], force_load_progress=True)
+        trainer.callbacks = [cb for cb in trainer.callbacks if not isinstance(cb, BatchSizeFinder)]
+
+    def on_fit_start(self, trainer, pl_module):
+        self.scale_batch_size(trainer, pl_module)
+
+        if self.early_exit:
+            self.pre_early_exit(trainer)
+        else:
+            trainer.callbacks = [cb for cb in trainer.callbacks if not isinstance(cb, BatchSizeFinder)]
+
+    def on_validation_start(self, trainer, pl_module):
+        if trainer.sanity_checking or trainer.state.fn != TrainerFn.VALIDATING:
+            return
+
+        self.scale_batch_size(trainer, pl_module)
+
+        if self.early_exit:
+            self.pre_early_exit(trainer)
+        else:
+            trainer.callbacks = [cb for cb in trainer.callbacks if not isinstance(cb, BatchSizeFinder)]
+
+    def on_test_start(self, trainer, pl_module):
+        self.scale_batch_size(trainer, pl_module)
+
+        if self.early_exit:
+            self.pre_early_exit(trainer)
+        else:
+            trainer.callbacks = [cb for cb in trainer.callbacks if not isinstance(cb, BatchSizeFinder)]
+
+    def on_predict_start(self, trainer, pl_module):
+        self.scale_batch_size(trainer, pl_module)
+
+        if self.early_exit:
+            self.pre_early_exit(trainer)
+        else:
+            trainer.callbacks = [cb for cb in trainer.callbacks if not isinstance(cb, BatchSizeFinder)]
+
+    def on_fit_end(self, trainer, pl_module):
+        if self.early_exit:
```

Comment:
```suggestion
        if not self.early_exit:
```

Reply:
Mind providing more information about the pre and post early exit. What are you trying to achieve?